File: android/app/src/main/java/com/billsafe/data/entities/Bill.kt
package com.billsafe.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity(tableName = "bills")
data class Bill(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val userId: String,
    val billName: String,
    val amount: Double,
    val dueDate: Int, // Day of month (1-31)
    val category: String, // electricity, water, phone, otp, subscription, etc.
    val frequency: String, // monthly, quarterly, yearly, one-time
    val description: String = "",
    val isActive: Boolean = true,
    val reminderDaysBefore: Int = 3,
    val lastPaidDate: LocalDateTime? = null,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
    val source: String = "manual" // manual or sms or email
)

File: backend/src/models/Bill.js
const mongoose = require('mongoose');

const billSchema = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    billName: {
        type: String,
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    dueDate: {
        type: Number,
        required: true,
        min: 1,
        max: 31
    },
    category: {
        type: String,
        enum: ['electricity', 'water', 'phone', 'credit_card', 'internet', 'subscription', 'other'],
        default: 'other'
    },
    frequency: {
        type: String,
        enum: ['monthly', 'quarterly', 'yearly', 'one-time'],
        default: 'monthly'
    },
    description: String,
    isActive: {
        type: Boolean,
        default: true
    },
    reminderDaysBefore: {
        type: Number,
        default: 3
    },
    lastPaidDate: Date,
    source: {
        type: String,
        enum: ['manual', 'sms', 'email'],
        default: 'manual'
    },
    createdAt: {
        type: Date,
        default: Date.now
    },
    updatedAt: {
        type: Date,
        default: Date.now
    }
});

// Index for efficient queries
billSchema.index({ userId: 1, dueDate: 1 });
billSchema.index({ userId: 1, isActive: 1 });

module.exports = mongoose.model('Bill', billSchema);
